modules/library/render/src/test/java/org/geotools/renderer/lite/gridcoverage2d/TransparencyStyledTest.java

```diff
@@ -0,0 +1,355 @@
+import org.geotools.gce.imagemosaic.ImageMosaicFormatFactory;
+import org.geotools.gce.imagemosaic.ImageMosaicReader;
+import org.geotools.referencing.CRS;
+import org.geotools.resources.image.ImageUtilities;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.RasterSymbolizer;
+import org.geotools.styling.Rule;
+import org.geotools.styling.SLDParser;
+import org.geotools.styling.Style;
+import org.geotools.styling.StyleFactory;
+import org.geotools.styling.StyledLayerDescriptor;
+import org.geotools.styling.UserLayer;
+import org.geotools.test.TestData;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.opengis.coverage.grid.GridCoverage;
+import org.opengis.parameter.GeneralParameterValue;
+import org.opengis.parameter.ParameterValue;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.NoSuchAuthorityCodeException;
+
+/**
+ * A Testing class involving footprint and transparency settings together with SLD, 
+ * making sure that transparency is preserved.
+ */
+public class TransparencyStyledTest {
+
+    private final static StyleFactory SF = CommonFactoryFinder
+            .getStyleFactory(GeoTools.getDefaultHints());
+
+    private GridCoverage2D readCoverage(File mosaicDirectory, FootprintBehavior fp, Color transparentColor)
+            throws NoSuchAuthorityCodeException, FactoryException, IOException {
+
+        ImageMosaicReader reader = (ImageMosaicReader) new ImageMosaicFormatFactory().createFormat()
+                .getReader(mosaicDirectory);
+
+        ParameterValue<String> footprintBehaviorParam = AbstractGridFormat.FOOTPRINT_BEHAVIOR
+                .createValue();
+        footprintBehaviorParam.setValue(fp.name());
+
+        ParameterValue<Color> inputTransparentColor = AbstractGridFormat.INPUT_TRANSPARENT_COLOR.createValue();
+        inputTransparentColor.setValue(transparentColor);
+
+        GeneralParameterValue[] readParams = new GeneralParameterValue[] { footprintBehaviorParam, inputTransparentColor};
+        GridCoverage2D coverage = reader.read(readParams);
+
+        reader.dispose();
+        assertNotNull(coverage);
+        return coverage;
+    }
+
+    /**
+     * Dispose the provided coverage for good.
+     * 
+     * @param coverage
+     */
+    private void disposeCoverage(GridCoverage2D coverage) {
+        if (coverage == null) {
+            return;
+        }
+        final RenderedImage im = coverage.getRenderedImage();
+        ImageUtilities.disposePlanarImageChain(PlanarImage.wrapRenderedImage(im));
+        coverage.dispose(true);
+
+    }
+
+    @AfterClass
+    public static void close() {
+        System.clearProperty("org.geotools.referencing.forceXY");
+        CRS.reset("all");
+    }
+
+    @BeforeClass
+    public static void init() {
+
+        // make sure CRS ordering is correct
+        CRS.reset("all");
+        System.setProperty("org.geotools.referencing.forceXY", "true");
```

Comment:
I know there is a bad habit of using this one "as string" and I'm also at fault for this, but would be nicer to use GeoTools.FORCE_LONGITUDE_FIRST_AXIS_ORDER